src/main/webapp/assets/script.js

```diff
@@ -72,7 +72,9 @@ function onLoad() {
   posts = fetchFakePosts();
   collegeLocation = fetchFakeCollegeLocation();
 
-  addPosts(posts);
+  if (document.getElementById('all-posts')) {
+    addPosts(posts);
+  }
```

Comment:
Ahh I encountered this too. Basically what is happening is the script is running before the 'all-posts' div has been added, I think. To fix this, I added "defer" to the script tag. I'll highlight that line in a comment.

Reply:
What was happening for me is that it's trying to run `addPosts()` on all pages, since `script.js` is being included in all pages (not all of which have an `all-posts` element), not just the feed page. I think maybe we might want to narrow the scope of script.js to possibly just include the feed page. What do you think?

Reply:
So accessing the home page was causing an error for me, but not the feed page

Reply:
Ooh I see! Yeah we should definitely talk about this in our daily standup today